src/components/popover/popover.test.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useRef, Fragment } from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Popover from './popover.jsx';
+
+describe('Popover', () => {
+  function PopoverWithToggle(popoverProps) {
+    const popoverRef = useRef();
+
+    return (
+      <Fragment>
+        <button onClick={() => { popoverRef.current.open = !popoverRef.current.open; }}>Toggle Popover</button>
+        <span>Some text</span>
+        <Popover ref={popoverRef} {...popoverProps}>{popoverProps.children}</Popover>
+      </Fragment>
+    );
+  }
+
+  it('has defaults', () => {
+    render(<Popover title="Defaults" />);
+    expect(document.body).toMatchSnapshot();
+  });
+
+  it('has a title', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Toggle Popover'));
+    expect(screen.getByText('Another title').tagName).toEqual('H1');
+  });
+
+  it('renders children', () => {
+    render(<PopoverWithToggle title="Another title"><span>More text</span></PopoverWithToggle>);
+
+    userEvent.click(screen.getByText('Toggle Popover'));
+    expect(screen.getByText('More text'));
+  });
+
+  it('sets left and top', () => {
+    render(<PopoverWithToggle title="Another title" left={100} top={200} />);
+
+    userEvent.click(screen.getByText('Toggle Popover'));
+    expect(screen.getByRole('dialog')).toHaveStyle('left: 100px');
+    expect(screen.getByRole('dialog')).toHaveStyle('top: 200px');
+  });
+
+  it('can be toggled open/closed', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Toggle Popover'));
+    expect(screen.getByText('Another title').tagName).toEqual('H1');
+
+    userEvent.click(screen.getByText('Toggle Popover'));
+    expect(screen.queryByText('Another title')).toEqual(null);
+  });
+
+  it('can be closed by clicking elsewhere in the window', () => {
+    render(<PopoverWithToggle title="Another title" />);
+
+    userEvent.click(screen.getByText('Toggle Popover'));
+    expect(screen.getByText('Another title').tagName).toEqual('H1');
```

Comment:
I think these test are trying to assert that the popover is on the page at this point? We want to avoid saying that the title is an H1 in case we change it in the future.

```suggestion
    expect(screen.getByText('Another title')).toBeInTheDocument()
```

Reply:
I do like that you skip this kind of assertion in the following tests -- we're basically saying that we _know_ the popover is on the page after clicking "Toggle Popover"

Maybe we want to do this same here?